### PR TITLE
feat(ui): Adding templating to code snippets

### PR DESCRIPTION
**Pull request by drdelambre**
This is a prereq for #14924 and wanted to make sure that it was applied separately, as it is its own chunk of functionality.

Basically it wraps a `CodeSnippet` with a small templating language based on a crowd favorite: https://johnresig.com/blog/javascript-micro-templating/

so anytime you submit a string of `Hello <%= planet %>`, it will go off and look up the property `planet` from the `value` prop passed into the component. if that's not defined, it'll look up the property in the `defaults` prop. if THAT is not defined, it just writes `undefined`.
``` Javascript
<TemplatedCodeSnippet
    template={ "Hello <%= planet %>" }
    values={{ planet: 'Mars' }}
    defaults={{ planet: 'World' }} />
```
^--- opens a CodeSnippet element with the text `Hello Mars`

### CODE REVIEW

File: ui/src/clientLibraries/components/ClientJavaOverlay.tsx [hunk 1]
```diff
@@ -30,24 +30,59 @@ const ClientJavaOverlay: FunctionComponent<{}> = () => {
...
+        }}
+      />
+      <TemplatedCodeSnippet template={pojoClassCodeSnippet} label="Java Code" />
       <h5>Execute a Flux query</h5>
-      <CodeSnippet copyText={executeQueryCodeSnippet} label="Java Code" />
+      <TemplatedCodeSnippet
+        template={executeQueryCodeSnippet}
+        label="Java Code"
+        defaults={{
+          bucket: 'my_bucket',
```
Comment (hoorayimhelping) [inline ui/src/clientLibraries/components/ClientJavaOverlay.tsx]: should this be `bucketID` and `orgID` for consistency?

File: ui/src/shared/components/TemplatedCodeSnippet.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  template: string
+  label: string
+  values?: StringMap
+  defaults?: StringMap
+}
+
+@ErrorHandling
+class TemplatedCodeSnippet extends PureComponent<Props> {
+  // NOTE: this is just a simplified form of the resig classic:
+  // https://johnresig.com/blog/javascript-micro-templating/
```
Comment (hoorayimhelping) [inline ui/src/shared/components/TemplatedCodeSnippet.tsx]: 🤙 dig this comment

File: ui/src/clientLibraries/components/ClientJavaOverlay.tsx (hunk 1 above)
Comment (drdelambre) [reply ui/src/clientLibraries/components/ClientJavaOverlay.tsx]: I was going to start standardizing those, but at the same time have a plan to connect them all to redux state in a little bit to give them MORE consistency. like site wide, frontend <=> backend consistency. so.... yes, but meh?

**Comment by drdelambre** [comment]
yeah, you're probably right. at least the templating function + object.assigns. ON IT

File: ui/src/clientLibraries/components/ClientJavaOverlay.tsx (hunk 1 above)
Comment (hoorayimhelping) [reply ui/src/clientLibraries/components/ClientJavaOverlay.tsx]: >so.... yes, but meh

pretty much perfectly sums up software engineering, circa 2019.

:+1:

File: ui/src/shared/components/TemplatedCodeSnippet.test.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+  it('should respect defaults', () => {
+    let props: any = {}
+    jest.mock('src/shared/components/CodeSnippet', () => _props => {
+      props = _props
+      return false
+    })
+
+    const template = 'this is only a <%= word %>'
+    const response = 'this is only a sentance'
```
Comment (hoorayimhelping) [inline ui/src/shared/components/TemplatedCodeSnippet.test.tsx]: defanately keep this in.

**Pull request merged**